### PR TITLE
Reset skip error counter on a fabric link if it was down.

### PR DESCRIPTION
**Pull request by jfeng-arista**
**What I did**
reset the skip error counters for fabric link monitoring if a link was down and up again. 

**Why I did it**

The fabric link monitoring feature does not take care the peer card restart cases. Peer card reload could lead to links go in to isolation state, and that's a case where we can ignore the error due to the init churn.  So in this change , the skip error on init counters get reset if a link was down and up again. 

**How I verified it**

**Details if related**

### CODE REVIEW

**Comment by kenneth-arista** [comment]
Related to https://github.com/sonic-net/sonic-buildimage/issues/19288

**Comment by judyjoseph** [comment]
@saksarav-nokia to review too

**Comment by saksarav-nokia** [comment]
Tested the fix and seems to be working. However i see lot of these log messages and not sure if we need have them at NOTICE level

2024 Aug  5 20:50:16.520538 ixre-cpm-chassis12 NOTICE swss0#orchagent: :- updateStateDbTable: PORT0 updates POLL_WITH_ERRORS to 0 0
2024 Aug  5 20:50:16.520538 ixre-cpm-chassis12 NOTICE swss0#orchagent: :- updateStateDbTable: PORT0 updates POLL_WITH_NO_ERRORS to 8 8
2024 Aug  5 20:50:16.520594 ixre-cpm-chassis12 NOTICE swss0#orchagent: :- updateStateDbTable: PORT0 updates POLL_WITH_FEC_ERRORS to 0 0
2024 Aug  5 20:50:16.520662 ixre-cpm-chassis12 NOTICE swss0#orchagent: :- updateStateDbTable: PORT0 updates POLL_WITH_NOFEC_ERRORS to 8 8
2024 Aug  5 20:50:16.520717 ixre-cpm-chassis12 NOTICE swss0#orchagent: :- updateStateDbTable: PORT0 updates CONFIG_ISOLATED to 0 0
2024 Aug  5 20:50:16.520782 ixre-cpm-chassis12 NOTICE swss0#orchagent: :- updateStateDbTable: PORT0 updates ISOLATED to 0 0

**Comment by jfeng-arista** [comment]
I can change the message  level to INFO

**Comment by jfeng-arista** [comment]
can anyone help review again and see if we can merge up this ? thank you

File: orchagent/fabricportsorch.cpp [hunk 1]
```diff
@@ -689,6 +706,43 @@ void FabricPortsOrch::updateFabricDebugCounters()
...
+        //        clear consecutivePollsWithNoErrors
+        //        clear consecutivePollsWithFecErrs
+        //        clear consecutivePollsWithNoFecErrs
+        //
+        //        if !MANUAL_ISOLATED && AUTO_ISOLATED :
+        //              autoIsolated = 0
+        //              isolated = 0
+        //        if MANUAL_ISOLATED:
+        //              autoIsolated = 0
+        //
```
Comment (arlakshm) [inline orchagent/fabricportsorch.cpp:723]: can you add this to the design document iso of here?

File: orchagent/fabricportsorch.cpp [hunk 2]
```diff
@@ -689,6 +706,43 @@ void FabricPortsOrch::updateFabricDebugCounters()
...
+        SWSS_LOG_INFO("Port %d lnk down cnt %d  handled: %d", lane, lnkDownCnt, preLnkDwnCnt);
+        if (lnkDownCnt != preLnkDwnCnt)
```
Comment (arlakshm) [inline orchagent/fabricportsorch.cpp:725]: when port goes down fabric port counters and isolation status will be cleared every polling cycle? Is this required?

Comment (arlakshm) [reply orchagent/fabricportsorch.cpp:725]: Do we need to clear the isolation status here?
should we wait for the  multiple polls without any errors after port comes up to unisolate the link?

File: orchagent/fabricportsorch.cpp [hunk 3]
```diff
@@ -1039,13 +1135,9 @@ void FabricPortsOrch::updateFabricCapacity()
         cur_event = "Higher";
     }
 
-    SWSS_LOG_NOTICE(" total link %d  expected link %d oper link %d event %s", total_links, expect_links, operating_links, cur_event.c_str());
```
Comment (arlakshm) [inline orchagent/fabricportsorch.cpp:1042]: why this log is removed?

**Review by arlakshm — COMMENTED** [review]
as comments

File: orchagent/fabricportsorch.cpp (hunk 2 above)
Comment (jfeng-arista) [reply orchagent/fabricportsorch.cpp:725]: > rt goes down fabric port counters and isolation status will be cleared every polling cycle? Is

Not every cycle, only when handle once when the linkdownCnt != handled linkDownCnt. 
Hard to tell from local information only if this is a new card come up or not, I would do this clean up , unless the link is shutdown by user, and monitoring starts from the beginning.

**Comment by jfeng-arista** [comment]
test_HashSwitchGlobalConfiguration[inner-frame-ecmp-hash] failed (1 runs remaining out of 2).
Not releated to this change

**Comment by jfeng-arista** [comment]
I don't quite get the coverage results, I added test to cover the change and get coverage tests passed before  retarting the pipeline. How it can fail the coverage test after the rerun ...

start again 

/azpw run Azure.sonic-swss

**Comment by arlakshm** [comment]
@prsunny, can you please help merge this PR

**Pull request merged**